backend/de.metas.async/src/main/java/de/metas/async/api/IAsyncBatchBL.java

```diff
@@ -110,7 +113,17 @@
 			@NonNull List<T> model,
 			@NonNull String asyncBatchInternalName);
 
+	IAutoCloseable assignTempAsyncBatchIdToModel(@NonNull Object model, @NonNull AsyncBatchId asyncBatchId);
+
 	I_C_Async_Batch getAsyncBatchById(AsyncBatchId asyncBatchId);
 
 	AsyncBatchId newAsyncBatch(String asyncBatchType);
+
+	Optional<String> getAsyncBatchTypeInternalName(@NonNull final I_C_Async_Batch asyncBatch);
+
+	boolean isAsyncBatchTypeInternalName(@NonNull I_C_Async_Batch asyncBatch, @NonNull String expectedInternalName);
```

Comment:
```suggestion
```

can be done later: i think we don't need them anymore..we can just get the `AsyncBatchType` now, right? And if needed, then `AsyncBatchType` should have this method

Reply:
they are for convenience. at least `isAsyncBatchTypeInternalName` is used in couple of places